### DISCUSSION TITLE
ShellCheck issue in VS Code

### DISCUSSION THREAD

**Discussion by Susja** [timonwong/vscode-shellcheck#1289] [Q&A] [answered]
Folks,
Please help to understand the workflow of extensions in VS Code.
I am using VS Code on Mac and Windows. I want to write bash script and would like to use 'helper-tool'. I found that ShellCheck extension could help me.
I downloaded VSIX and installed ShellCheck with no issue.
The issue is that it does not work for me ...
E.g. I write 'valid' bash script in VS Code but it is not recognized as shell script. If I copy/paste it on https://www.shellcheck.net/ it works perfectly fine BUT in VS Code I see this:
Language: `shellcheckrc`
# ShellCheck
 Version: `0.9.0`
 Bundled: `true`
# Warnings
 Document is not a shell script or is filtered out

Could someone help? Note: I don't have admin permission on Mac hence I had to download .vsix and install it.

Thanks

**Comment by felipecrs** [comment]
What is your file extension? You can also forcefully change the document language to shell script at the bottom right corner of vscode.

**Comment by Susja** [comment]
hello @felipecrs 

First off ... I installed it on Windows and have the same error.
here's my file:
--
#!/bin/bash

a="Hello"
b="World"

if [ $a == "Hello" ] ; then 
echo $a $b
-----
I saved it as test.sh
I put some errors on purpose in the file.
When I try to execute it in terminal it obviously fails with these errors:
--
./test1.sh: line 8: syntax error: unexpected end of file
--
Is it what one should expect from this extension?
When I run it in https://www.shellcheck.net/ ... it gives me 'expected' explanation of error.
Hence, extension does not work in VS Studio neither Mac nor Windows in my case.
What am I doing wrong?
Thanks again ...
P.S. I tried to change the document language at the bottom right corner to shellcheck-output but it did not make a difference. How/when shellcheck should evaluate the script: when one writes it or when execute? On web it works perfectly i.e. it checks while I'm writing ... Maybe  I am missing something?

**Reply by felipecrs** [reply, marked as answer]
The correct document language should be Shell Script. Also, check your file line endings, since you mentioned Windows, I suspect you have incorrect line endings (CRLF as opposed to LF).

**Reply by felipecrs** [reply]
There is also a GIF in the README where you can see what is expected.

**Comment by Susja** [comment]
@felipecrs 
I checked that I don't have in VS language Shell Script. 
Well ... actually I'm not using VS Code. I am using ADS. Maybe that's why I don't see 'shell script' language? Maybe that's why I have an issue with ShellChecker?

Thanks

**Reply by felipecrs** [reply]
Yeah, it could be that. Make sure to get "Shell Script" language working first.

**Comment by Susja** [comment]
Thanks for the tip!
